core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/StructuredStreamingPlugin.scala

```diff
@@ -0,0 +1,109 @@
+      }
+  }
+
+  override val writeNodeProcessor: PartialFunction[LogicalPlan, WriteNodeInfo] = {
+    case `_: WriteToDataSourceV2`(writeToDataSource) =>
+      val writer = extractFieldValue[AnyRef](writeToDataSource, "writer")
+      val query = extractFieldValue[LogicalPlan](writeToDataSource, "query")
+
+      writer match {
+        case `_: MicroBatchWriter`(microBatchWriter) =>
+          val dataSourceWriter = extractFieldValue[AnyRef](microBatchWriter, "writer")
+          dataSourceWriter match {
+
+            case `_: KafkaStreamWriter`(kafkaStreamWriter) =>
+              val topic = extractFieldValue[Option[String]](kafkaStreamWriter, "topic")
+              val producerParams = extractFieldValue[Map[String, String]](kafkaStreamWriter, "producerParams")
+              val bootstrapServers = producerParams("bootstrap.servers")
+
+              (SourceIdentifier(Some("kafka"), asURI(topic.get)), SaveMode.Append, query, Map("bootstrap.servers" -> bootstrapServers))
+
+            case `_: FileStreamWriter`(fileStreamWriter) =>
+              val fileStreamSink = extractFieldValue[FileStreamSink](fileStreamWriter, "fileStreamSink")
+              val path = extractFieldValue[String](fileStreamSink, "path")
+              val fileFormat = extractFieldValue[DataSourceRegister](fileStreamSink, "fileFormat")
+              val options = extractFieldValue[Map[String, String]](fileStreamSink, "options")
+
+              (SourceIdentifier(Some(fileFormat.shortName()), asURI(path)), SaveMode.Append, query, options)
+          }
+      }
+  }
+}
+
+object StructuredStreamingPlugin {
+  object `_: StreamingDataSourceV2Relation` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.execution.datasources.v2.StreamingDataSourceV2Relation")
+
+  object `_: KafkaMicroBatchReader` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.kafka010.KafkaMicroBatchReader")
+
+  object `_: KafkaStreamWriter` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.kafka010.KafkaStreamWriter")
+
+  object `_: WriteToDataSourceV2` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2")
+
+  object `_: MicroBatchWriter` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.execution.streaming.sources.MicroBatchWriter")
+
+  object `_: FileStreamWriter` extends SafeTypeMatchingExtractor[AnyRef](
+    "za.co.absa.spline.harvester.plugin.embedded.FileStreamWriter")
+
+  private def asURI(topic: String) = s"kafka:$topic"
+}
+
+// https://issues.apache.org/jira/browse/SPARK-27484
+class FileStreamWriter(val fileStreamSink: FileStreamSink) extends StreamWriter {
+  override def commit(epochId: Long, messages: Array[WriterCommitMessage]): Unit = {}
+
+  override def abort(epochId: Long, messages: Array[WriterCommitMessage]): Unit = {}
+
+  override def createWriterFactory(): DataWriterFactory[InternalRow] = null
```

Comment:
How about throwing `UnsupportedOperationException` instead of returning `null`?

Reply:
Yop

Reply:
ok